force-app/main/default/classes/ServiceSelector_TEST.cls

```diff
@@ -74,4 +74,92 @@ public with sharing class ServiceSelector_TEST {
         }
         Test.stopTest();
     }
+
+    @IsTest
+    private static void testGetServicesByProgramEngagementId() {
+        ProgramTestDataFactory.insertTestData(true);
+        Program__c program1 = [
+            SELECT Id, Name
+            FROM Program__c
+            WHERE Name = 'Program 1'
+            LIMIT 1
+        ][0];
+
+        ProgramEngagement__c engagement1 = [
+            SELECT Id, Name
+            FROM ProgramEngagement__c
+            WHERE Program__c = :program1.Id
+            LIMIT 1
+        ];
+
+        Set<Id> expectedIds = (new Map<Id, Service__c>(
+                [
+                    SELECT Id, Name, Program__c
+                    FROM Service__c
+                    WHERE
+                        Program__c IN (
+                            SELECT Program__c
+                            FROM ProgramEngagement__c
+                            WHERE Id = :engagement1.Id
+                        )
+                ]
+            ))
+            .keySet();
+
+        Test.startTest();
+        ServiceSelector selector = new ServiceSelector();
+        List<Service__c> actual = selector.getServicesByProgramEngagementId(
+            engagement1.Id
+        );
+        Test.stopTest();
+
+        System.assert(!expectedIds.isEmpty());
```

Comment:
Messages are totally not required but I find them super helpful especially on this one because the only thing we will see if it fails, is that it failed with no clues without going to the code :)

Reply:
Sure.. I can add them it is not a problem at all